src/hooks/useFollowScroll.ts

```
@@ -0,0 +1,66 @@
+import { useRef, useEffect, useState, useCallback } from "react";
+import useDeviceSize from "./useDeviceSize";
+
+export default function useFollowScroll<T extends HTMLElement>(startFollowPosition = 200, damping = 0.1) {
+  const elementRef = useRef<T | null>(null);
+  const targetPosition = useRef(0);
+  const currentPosition = useRef(0);
+  const [initialPosition, setInitialPosition] = useState<number | null>(null);
+  const deviceSize = useDeviceSize();
+
+  useEffect(() => {
+    elementRef.current && (elementRef.current.style.willChange = "transform");
```

Biome

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)